src/main/java/com/google/devtools/build/lib/vfs/OsPathPolicy.java

```diff
@@ -135,4 +144,15 @@ static int removeRelativePaths(String[] segments, int starti, boolean isAbsolute
       return segmentCount;
     }
   }
+
+  /**
+   * Unchecked exception thrown by {@link OsPathPolicy} implementations when a path cannot be
+   * normalized on the current host OS.
+   */
+  final class UncheckedPathUnsupportedOnThisOsException
+      extends UnsupportedOperationException {
```

Comment:
does this need to be an unchecked exception? It looks like it's only thrown from `ShortPathResolver.resolve()`, which itself is only called through `OsPathPolicy` from `PathFragment()` so it should not be too bad to either use its checked version or turn this into an erroneous return value instead.

Reply:
Unfortunately this is also reachable via `PathFragment#getRelative`, which has ~100 callsites in prod code in Bazel.

But as `PathFragment#getRelative` can validly throw the exception (e.g. `C:\foobar` is joined with `foo~1.txt` on Linux for Windows), this could also result in the unchecked exception being thrown when C++ toolchain logic internally uses this function. I'm not sure what a good solution to this would look like.